common/src/main/java/com/wynntils/models/abilities/ArrowShieldModel.java

```diff
@@ -117,9 +117,11 @@ public int getArrowShieldCharge() {
     }
 
     private void registerShield() {
-        spawnedArrowIds = collectedArrowIds;
-        collectedArrowIds = null;
-        WynntilsMod.postEvent(new ArrowShieldEvent.Created(spawnedArrowIds.size()));
+        if (collectedArrowIds != null) {
```

Comment:
Is this the root cause? Is there not a race somewhere? I saw the exception log too, saw tick scheduler. Maybe we have a wrong assumption somewhere?

Reply:
It is possible that the cast is triggered multiple times within 400 seconds causing the collected list to become null then cascading to spawnedArrowIds, but this should catch that. There's no other instance of collectedArrowIds becoming null